configure.ac

```diff
@@ -1021,7 +1052,9 @@ M0_CPPFLAGS_DIST=$(echo "$M0_CPPFLAGS" | grep -Po -e '-D\S+|-I/usr\S+' | xargs e
 # __attribute__, such as __attribute__((gccxml(...))). It's important because we
 # use -Werror, which turns warnings into errors, but we still need to be able to
 # use gccxml attributes.
-M0_CFLAGS="-pipe -Wall -Werror -Wno-attributes -fno-strict-aliasing $M0_CFLAGS"
+PAGE_SIZE=$(getconf PAGE_SIZE)
+PAGE_SHIFT=$(( $((PAGE_SIZE==4096)) ? 12 :  $(($((PAGE_SIZE==65536)) ? 16 :  14)) ))
```

Comment:
Why it is fragile ? either we have to include the page size definition in  sys/user.h  file (in linux include file)  as it is missing in aarch64 platform
or we have to follow the above code for the time being

Reply:
if  I include the follwing statement in /usr/include/sys/user.h  file
#define PAGE_SHIFT 16
#define PAGE_SIZE  (1UL << PAGE_SHIFT)

it is compiling

Reply:
According to the [man-page](https://man7.org/linux/man-pages/man2/getpagesize.2.html), this is the proper way to get the page size:

```
#include <unistd.h>
           long sz = sysconf(_SC_PAGESIZE);
```

Reply:
@nikitadanilov, updated with your suggestion of automatic PAGE_SHIFT calculation via logarithm and bc.

Reply:
@Nalini-Seagate: it is fragile, because this particular expression tacitly assumes that the only valid page sizes are 4K, 16K and 64K and will silently produce an incorrect result if a different page size is used.

@andriytk: in user space page size is not very useful (it only defines mmap granularity). In our user space code we do not need actual page size, we just need *some* convenient buffer size and PAGE_SIZE is used because the code is ahared with kernel or was converted from kernel code.

Reply:
@nikitadanilov I absolutely agree. I only suggest that this improvement (getting rid of PAGE_SIZE in user-space code) to be done as a separate task in a different PR (as it might take some time). What do you think?

Reply:
@nikitadanilov I tried it in a separate PR, please see https://github.com/Seagate/cortx-motr/pull/1109.